packages/venia-concept/src/components/Checkout/braintreeDropin.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * @fileoverview This component uses the BrainTree Web Drop In to
+ * hook into Web Payments and the Payment Request API to
+ * submit payments via BrainTree.
+ *
+ * See:
+ * https://github.com/braintree/braintree-web-drop-in
+ * https://braintree.github.io/braintree-web-drop-in/docs/current/index.html
+ * https://developers.braintreepayments.com/guides/drop-in/overview/javascript/v3
+ */
+
+import React, { Component } from 'react';
+import { bool, func } from 'prop-types';
+import dropin from 'braintree-web-drop-in';
+
+const BRAINTREE_SANDBOX_TOKEN = 'sandbox_8yrzsvtm_s2bg8fs563crhqzk';
+
+class BraintreeDropin extends Component {
+    static propTypes = {
+        isRequestingPaymentNonce: bool,
+        onError: func.isRequired,
+        onSuccess: func.isRequired
+    };
+
+    state = {
+        isSupported: true
+    };
+
+    componentDidMount() {
+        this.createDropinInstance()
+            .then(instance => {
+                this.braintree = instance;
+            })
+            .catch(() => {
+                this.setState({ isSupported: false });
+            });
+    }
+
+    componentWillUnmount() {
+        this.braintree = null;
+    }
+
+    componentDidUpdate(prevProps) {
+        if (
+            this.braintree &&
+            this.props.isRequestingPaymentNonce &&
+            !prevProps.isRequestingPaymentNonce
+        ) {
+            // Our parent is telling us to request the payment nonce.
+            this.requestPaymentNonce();
+        }
+    }
```

Comment:
I'm open for comments on this approach. The parent needed some way to trigger this call on the child and I was trying to prefer a declarative approach to an imperative one.

Reply:
Another approach would be to surface the instance's availability to the wrapping form like this guy did [here](https://github.com/Cretezy/braintree-web-drop-in-react). Check out `onInstance` and how he ties the dropin to the form. Kind of how you did but with `onSuccess`.

Reply:
Dang good find I didn't even see that. Perhaps this can be a future thing, I think ours is fine for now.

Reply:
This kind of workflow will get much simpler with hooks very shortly. As long as this works as expected, it should be fine for now.